### ISSUE TITLE
reconsider gRPC

### ISSUE THREAD

**Issue by v0lkan** [shieldworks/aegis#266]
I mean, this is not a full-duplex streaming app.

The API merely passes a bunch of key-value pairs back and forth.

Everything runs in the same cluster, things are relatively fast.

Even when there is a network error, the client can retry the command; gRPC’s auto retry mechanism proves little benefit.

REST is easier to document; simpler, and a larger audience can be comfortable with a RESTful API.

We won’t need to deal with auto-generating stubs.

I belive for this current use-case, gRPC is an overkill.

We can consider it if there is a need; and maybe implement a gRPC API later.

Had we started with gRPC, it might have made sense to go with that way, but Aegis Safe already started with REST, so better to stick with it.

**Comment by v0lkan** [comment]
I will think about this with a clearer mind; and depending on how things go, I might close gRPC-related issues.

**Comment by v0lkan** [comment]
better to continue with the RESTful model for now.

rest is async, simple, and a perfect fit for this use-case.

We don't need streams; we don't need full duplex communication; and the typical payload of a secret is not too much, clients are not dependent on a extremely fast response, data compression is not a big deal either. there is not too much of a parallelism need; and there is no need to reuse a connection to transfer multiple stuff.

In short, although gRPC has its advantages; the benefits are not significant enough to make a leap.

Closing it for now; we can reopen related gPRC ticket such as #97 when we really think that there is a need
